src/flag_gems/ops/slice_scatter.py

```diff
@@ -2,584 +2,39 @@
-        B,
-        A_out,
-        N1,
-        N2,
-        K,
-        am_stride,
-        an_stride,
-        ak_stride,
-        bm_stride,
-        bn_stride,
-        bk_stride,
-        start,
-        end,
-        step,
-        BLOCK=512,
-    )
-
-
-@triton.jit(
-    do_not_specialize=[
-        "N1",
-        "N2",
-        "K",
-        "am_stride",
-        "an_stride",
-        "ak_stride",
-        "bm_stride",
-        "bn_stride",
-        "bk_stride",
-        "start",
-        "end",
-        "step",
-    ]
-)
-def scatter_3d_mid_kernel(
-    A,
-    B,
-    A_out,
-    # mid dim size of A and A_out
-    N1,
-    # mid dim size of B
-    N2,
-    # inner dim size of inp, src and out
-    K,
-    # strides of A and A_out
-    am_stride,
-    an_stride,
-    ak_stride,
-    # strides of B
-    bm_stride,
-    bn_stride,
-    bk_stride,
-    # slice start index
-    start,
-    # slice end index
-    end,
-    # slice step
-    step,
-    NBLOCK: tl.constexpr,
-    KBLOCK: tl.constexpr,
-):
-    # Each cta processes one [1, NBLOCK, KBLOCK] chunk for input and output
-    # The src chunk size is dynamically determined
-
-    pidx = tle.program_id(0)
-    pidy = tle.program_id(1)
-
-    am_idx = pidx
-    an_idx = pidy * NBLOCK
-
-    # Offsets into inp and out chunks
-    n_idx_tile = an_idx + tl.arange(0, NBLOCK)[:, None]
-    k_idx_tile = tl.arange(0, KBLOCK)[None, :]
-    a_offset = am_idx * am_stride + n_idx_tile * an_stride + k_idx_tile * ak_stride
-    n_mask = n_idx_tile < N1
-    k_mask = k_idx_tile < K
-
-    # Offsets into src
-    b_offset = am_idx * bm_stride
-    b_offset += (n_idx_tile - start) * bn_stride // step
-    b_offset += k_idx_tile * bk_stride
-    # This mask applies to the [NBLOCK, KBLOCK]
-    bn_mask = start <= n_idx_tile
-    bn_mask &= n_idx_tile < end
-    bn_mask &= (n_idx_tile - start) % step == 0
-
-    # merge inp and src then write back
-    inp = tl.load(A + a_offset, mask=n_mask & k_mask)
-    src = tl.load(B + b_offset, mask=bn_mask & k_mask)
-    out = tl.where(bn_mask & k_mask, src, inp)
-    tl.store(A_out + a_offset, out, mask=n_mask & k_mask)
-    # tl.store(A_out + a_offset, src, mask=bn_mask & k_mask)
-
-
-@libentry()
-@triton.autotune(
-    configs=[
-        triton.Config(kwargs={"R": 1, "C": 512}, num_warps=4),
-        triton.Config(kwargs={"R": 32, "C": 32}, num_warps=4),
-        triton.Config(kwargs={"R": 64, "C": 64}, num_warps=4),
-        triton.Config(kwargs={"R": 4, "C": 512}, num_warps=4),
-        triton.Config(kwargs={"R": 16, "C": 128}, num_warps=4),
-    ],
-    key=["strided", "pivoted"],
-)
-@triton.heuristics(
-    values={
-        "predicate_load": lambda args: args["x_ncol"] > 0.5 * args["y_ncol"],
-    }
-)
-@triton.jit(
-    do_not_specialize=[
-        "nrow",
-        "x_ncol",
-        "y_ncol",
-        "x_si",
-        "x_sj",
-        "y_si",
-        "y_sj",
-        "start",
-        "end",
-        "step",
-    ]
-)
-def scatter_2d_inner_kernel(

```

Comment:
sad story: having to delete so much code

Reply:
But it's a better move!